src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,36 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+
+	
+	public String value;
+	protected boolean isPrivate;
+	
+	public Contact(boolean isPrivate){
+		this.isPrivate = isPrivate;
+	}
+	
+	public boolean isValid(String value, String regex) {
+		return value.matches(regex);
+	}
+	
+	@Override
+	public String toString(){
+		return value;
+	}
+	
+	@Override
+	public int hashCode(){
+		return value.hashCode();
+	}
+	
+    public boolean isPrivate() {
```

Comment:
Use naming conventions for common methods. Therefore, this method should be called `getIsPrivate`. Eclipse can automatically generate the getter or/and setter function for you.